pkg/util/v1alpha2/katibclient/katib_client.go

```diff
@@ -20,18 +20,40 @@ import (
 	"io/ioutil"
 	"strings"
 
+	"k8s.io/client-go/rest"
+
 	experimentsv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
+type Client interface {
+	GetExperimentList(namespace ...string) (*experimentsv1alpha2.ExperimentList, error)
+	CreateExperiment(experiment *experimentsv1alpha2.Experiment, namespace ...string) error
+	GetConfigMap(name string, namespace ...string) (map[string]string, error)
+	GetTrialTemplates(namespace ...string) (map[string]string, error)
+	UpdateTrialTemplates(newTrialTemplates map[string]string, namespace ...string) error
+	GetMetricsCollectorTemplates(namespace ...string) (map[string]string, error)
+	UpdateMetricsCollectorTemplates(newMCTemplates map[string]string, namespace ...string) error
+}
+
 type KatibClient struct {
 	client client.Client
 }
 
-func NewClient(options client.Options) (*KatibClient, error) {
+func MustNewTestClient(cfg *rest.Config) Client {
```

Comment:
Move test code to a common folder for tests ?

Reply:
Actually, it is not only for test purpose, let me refactor the code here.

Reply:
The code is not needed now since we make client a mockable interface, thus we can delete it safely.